### ISSUE TITLE
TOKEN_PRIVILEGES

### ISSUE THREAD

**Issue by kennykerr** [microsoft/win32metadata#364]
This struct is defined as follows:

```C++
typedef struct _TOKEN_PRIVILEGES {
    DWORD PrivilegeCount;
    LUID_AND_ATTRIBUTES Privileges[ANYSIZE_ARRAY];
} TOKEN_PRIVILEGES, *PTOKEN_PRIVILEGES;
```

However the metadata emits it as follows:

```C#
public struct TOKEN_PRIVILEGES
{
	public uint PrivilegeCount;
	public TOKEN_PRIVILEGES_PrivilegesFlags Privileges;
}
```

There are two main issues:

* How to deal with the trailing variable sized array. 
* The type of the second field is incorrect.

**Comment by sotteson1** [comment]
@AArnott, the scraper thought the flags should go on the Privileges member of TOKEN_PRIVILEGES, but it should have gone on LUID_AND_ATTRIBUTES.Attributes. If you plan on re-running the docs enum scraper, maybe you need to add a hint or tell it not to scrape this page or something else.

As for the second issue, we've talked about adding an attribute in the metadata to arrays to indicate the arrays are variable length. @kennykerr, would that solution work for you?

**Comment by kennykerr** [comment]
Yes, an attribute sounds good - thanks!

**Comment by AArnott** [comment]
#266 tracks the attribution of variable-length arrays.
Yes, I expect to run the scraper at least once more. I'll copy this bug to track the issue in the scraper, assuming you'll fix the broken API in the meantime.

**Comment by sotteson1** [comment]
This has been fixed, but still need to do #266 to add the attribute to variable length arrays.
